src/StimulusBundle/src/Form/Extension/FormTypeExtension.php

```diff
@@ -0,0 +1,162 @@
+            }
+
+            $attributes = array_merge($view->vars['attr'], $this->stimulusAttributes->toArray());
+            $view->vars['attr'] = $attributes;
+        }
+
+        foreach (['row_attr', 'choice_attr'] as $index) {
+            if (
+                isset($options[$index])
+                && (
+                    isset($options[$index]['stimulus_controller'])
+                    || isset($options[$index]['stimulus_target'])
+                    || isset($options[$index]['stimulus_action'])
+                )
+            ) {
+                $this->stimulusAttributes = new StimulusAttributes(new Environment(new ArrayLoader()));
+
+                if (isset($options[$index]['stimulus_controller'])) {
+                    $this->handleController($options[$index]['stimulus_controller']);
+                    unset($options[$index]['stimulus_controller']);
+                }
+
+                if (isset($options[$index]['stimulus_target'])) {
+                    $this->handleTarget($options[$index]['stimulus_target']);
+                    unset($options[$index]['stimulus_target']);
+                }
+
+                if (isset($options[$index]['stimulus_action'])) {
+                    $this->handleAction($options[$index]['stimulus_action']);
+                    unset($options[$index]['stimulus_action']);
+                }
+
+                $attributes = array_merge($options[$index], $this->stimulusAttributes->toArray());
+                $view->vars[$index] = $attributes;
+            }
+        }
+    }
+
+    private function handleController(string|array $controllers): void
+    {
+        if (\is_string($controllers)) {
+            $controllers = [$controllers];
+        }
+
+        foreach ($controllers as $controllerName => $controller) {
+            if (\is_string($controller)) { // 'stimulus_controller' => ['controllerName1', 'controllerName2']
+                $this->stimulusAttributes->addController($controller);
+            } elseif (\is_array($controller)) { // 'stimulus_controller' => ['controllerName' => ['values' => ['key' => 'value'], 'classes' => ['key' => 'value'], 'targets' => ['otherControllerName' => '.targetName']]]
+                $this->stimulusAttributes->addController((string) $controllerName, $controller['values'] ?? [], $controller['classes'] ?? [], $controller['outlets'] ?? []);
+            }
+        }
+    }
+
+    private function handleTarget(array $targets): void
+    {
+        foreach ($targets as $controllerName => $target) {
+            $this->stimulusAttributes->addTarget($controllerName, \is_array($target) ? implode(' ', $target) : $target);
+        }
+    }
+
+    private function handleAction(string|array $actions): void
+    {
+        // 'stimulus_action' => 'controllerName#actionName'
+        // 'stimulus_action' => 'eventName->controllerName#actionName'
+        if (\is_string($actions) && str_contains($actions, '#')) {
```

Comment:
The reason i want another object, we should not have to duplicate this in multiple place. Will open a draft later if i can

Reply:
(i cannot 😅 🕐 )